src/github/github.installation.controller.ts

```diff
@@ -0,0 +1,203 @@
+import {
+  Controller,
+  Get,
+  Post,
+  Put,
+  Delete,
+  Body,
+  Param,
+  Query,
+  HttpException,
+  HttpStatus,
+} from '@nestjs/common';
+import { AllowedOrganization } from './entities/allowed-organization.entity';
+import { OrganizationStatus } from 'src/common/constants';
+import { GithubInstallationService } from './github.installation.service';
+import { ApiTags } from '@nestjs/swagger';
+import { CreateOrgDto } from './dto/create.org.dto';
+
+@ApiTags('Github-Installation')
+@Controller('github-installation')
+export class GithubInstallationController {
+  constructor(
+    private readonly githubInstallationService: GithubInstallationService,
+  ) {}
+
+  @Post()
+  async create(
+    @Body() createOrgDto: CreateOrgDto,
+  ): Promise<AllowedOrganization> {
+    try {
+      return this.githubInstallationService.create(createOrgDto);
+    } catch (error) {
+      if (error.status) {
+        throw new HttpException(error.message, error.status);
+      } else {
+        throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
+      }
+    }
+  }
+
+  @Get()
+  async findAll(): Promise<AllowedOrganization[]> {
+    try {
+      return this.githubInstallationService.findAll();
+    } catch (error) {
+      if (error.status) {
+        throw new HttpException(error.message, error.status);
+      } else {
+        throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
+      }
+    }
+  }
+
+  @Get(':id')
+  async findOne(@Param('id') id: string): Promise<AllowedOrganization> {
+    try {
+      return this.githubInstallationService.findOne(id);
+    } catch (error) {
+      if (error.status) {
+        throw new HttpException(error.message, error.status);
+      } else {
+        throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
+      }
+    }
+  }
+
+  @Get('by-name/:organizationName')
+  async findByName(
+    @Param('organizationName') organizationName: string,
+  ): Promise<AllowedOrganization> {
+    try {
+      return this.githubInstallationService.findByName(organizationName);
+    } catch (error) {
+      if (error.status) {
+        throw new HttpException(error.message, error.status);
+      } else {
+        throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
+      }
+    }
+  }
+
+  @Put(':id')
+  async update(
+    @Param('id') id: string,
+    @Body() updateOrgDto: CreateOrgDto,
+  ): Promise<AllowedOrganization> {
+    try {
+      return this.githubInstallationService.update(id, updateOrgDto);
+    } catch (error) {
+      if (error.status) {
+        throw new HttpException(error.message, error.status);
+      } else {
+        throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
+      }
```

Comment:
PullProofAI

Refactor the error handling code to a shared method to avoid code duplication.
```diff
-     try {
-       return this.githubInstallationService.update(id, updateOrgDto);
-     } catch (error) {
-       if (error.status) {
-         throw new HttpException(error.message, error.status);
-       } else {
-         throw new HttpException(error.message, HttpStatus.BAD_REQUEST);
-       }
-     }
+     return this.handleServiceError(() => this.githubInstallationService.update(id, updateOrgDto));
```



<!-- This is an auto-generated comment by PullProofAI -->